lib/matplotlib/axes/_axes.py

```diff
@@ -6054,25 +6054,30 @@ def _normalize_input(inp, ename='input'):
             """
             if (isinstance(x, np.ndarray) or
                     not iterable(cbook.safe_first_element(inp))):
-                # TODO: support masked arrays;
-                inp = np.asarray(inp)
+
+                inp = cbook.safe_masked_invalid(inp)
                 if inp.ndim == 2:
                     # 2-D input with columns as datasets; switch to rows
                     inp = inp.T
+
+                    if inp.shape[1] < inp.shape[0]:
```

Comment:
Absent a compelling reason I err on the side of not removing things (which may not be a fully defend able position, but it is a safe conservative one)

Chasing this back through git blame, the warning came into the code base in
214976f758f91ed8a534b6f5f517a9aebbe8e3fb in 2008.